discovery/docker_discovery_test.go

```diff
@@ -197,5 +218,48 @@ func Test_DockerDiscovery(t *testing.T) {
 				So(container, ShouldBeNil)
 			})
 		})
+
+		Convey("Run()", func() {
+			disco.sleepInterval = 1 * time.Millisecond
+
+			Convey("pings Docker", func() {
+				disco.Run(&dummyLooper{})
+
+				// Check a few times that it tries to ping Docker
+				for i := 0; i < 3; i++ {
+					pinged := false
+					select {
+					case <-client.PingChan:
+						pinged = true
+					case <-time.After(100 * time.Millisecond):
```

Comment:
I'd make this smaller so that in the case of failure it bombs out faster. 10 millis is probably fine.